### PR TITLE
C2 karimngr c5s08 09

### PR DESCRIPTION
**Pull request by ghost**
#349

the story is done please review according to the scenarios found in the backlog

@AmrARaouf @Omar-Hossam @Mirna-Yacout 
@m0hamed

### CODE REVIEW

File: arability/app/views/admin/list_developer_projects.html.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<div class="container admin">
+  <div class="row">
+    <%= render partial: "sidebar-nav", locals: {active: nil} %>
+    <div class="span9 offset3">
+      <div class="tab-content">
+        <div class="tab-pane well span7 active">
+          <legend>
+            <div class="pull-left pagination">
+              <%= paginate @list, :remote => true %>
```
Comment (omar-mohyeldin) [inline arability/app/views/admin/list_developer_projects.html.erb]: avoid using =>

**Comment by omar-mohyeldin** [comment]
@karimngr almost verified, just please adjust this small mistake

**Comment by ghost** [comment]
@Omar-Hossam please review

File: arability/app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -52,23 +52,88 @@ def list_prizes
     @prizes_list = Prize.order(params[:order]).page(params[:page]).per(2)
   end
 
+  # Author:
+  #   Karim ElNaggar
+  # Description:
+  #   this action lists the gamers and links to make them admins
+  # Params
+  #  none
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb]: how are the params none when you clearly use params[:order] and params[:page]

File: arability/app/controllers/admin_controller.rb [hunk 3]
```diff
@@ -52,23 +52,88 @@ def list_prizes
...
   def list_gamers
     @list = Gamer.order(params[:order]).page(params[:page]).per(5)
   end
 
+  # Author:
+  #   Karim ElNaggar
+  # Description:
+  #   this action lists the developers
+  # Params
+  #  none
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb]: same comment as the previous comment

File: arability/app/controllers/admin_controller.rb [hunk 4]
```diff
@@ -52,23 +52,88 @@ def list_prizes
...
   def list_developers
     @list = Developer.order(params[:order]).page(params[:page]).per(5)
   end
 
+  # Author:
+  #   Karim ElNaggar
+  # Description:
+  #   this action lists the admins and links to remove them
+  # Params
+  #  none
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb]: and again

File: arability/app/controllers/admin_controller.rb [hunk 5]
```diff
@@ -52,23 +52,88 @@ def list_prizes
...
     @list = Gamer.where(:admin => true).order(params[:order])
                           .page(params[:page]).per(5)
   end
 
+  # Author:
+  #   Karim ElNaggar
+  # Description:
+  #   this action lists the projects
+  # Params
+  #  none
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb]: and again

File: arability/app/controllers/admin_controller.rb [hunk 6]
```diff
@@ -52,23 +52,88 @@ def list_prizes
...
   def list_projects
     @list = Project.order(params[:order]).page(params[:page]).per(5)
   end
 
+  # Author:
+  #   Karim ElNaggar
+  # Description:
+  #   this action lists the projects of a user
+  # Params
+  #   developer_id: the id of the developer
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb:123]: add the two params :order and :page

File: arability/app/controllers/admin_controller.rb [hunk 7]
```diff
@@ -52,23 +52,88 @@ def list_prizes
...
+  #   Karim ElNaggar
+  # Description:
+  #   this action makes a gamer an admin
+  # Params
+  #  none
+  # Success: 
+  #   redirects the new admin to list admins view and display success message
+  # Failure: 
+  #   the user is not a valid user so an error is displayed
   def make_admin
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb:145]: please add a function in the gamer model that deals with accessing the database

File: arability/app/controllers/admin_controller.rb [hunk 8]
```diff
@@ -83,6 +148,16 @@ def make_admin
...
+  #   Karim ElNaggar
+  # Description:
+  #   this action removes an admin from the admins
+  # Params
+  #  none
+  # Success: 
+  #   removes the admin and redirects to list admins view
+  # Failure: 
+  #   the user is not a valid admin so an error is displayed
   def remove_admin
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb:168]: please add a function in the gamer model that deals with accessing the database

**Comment by AmrARaouf** [comment]
@karimngr please re-modify docs

**Comment by AmrARaouf** [comment]
@karimngr the tests are fine but please add the tests for the model functions discussed in the comments

**Comment by ghost** [comment]
@Omar-Hossam @AmrARaouf @Mirna-Yacout  please review

File: arability/app/models/gamer.rb [hunk 9]
```diff
@@ -62,6 +62,38 @@ def self.find_for_database_authentication(gamer_conditions)
   end
 
   # Description:
+  #   makes the user an admin
+  # Author:
+  #   Karim el naggar
+  # @params:
```
Comment (AmrARaouf) [inline arability/app/models/gamer.rb]: Params capital

File: arability/app/models/gamer.rb [hunk 10]
```diff
@@ -62,6 +62,38 @@ def self.find_for_database_authentication(gamer_conditions)
...
+  #   user: the user to be converted
+  # returns:
```
Comment (AmrARaouf) [inline arability/app/models/gamer.rb]: no "return"
Success (capital S)
Failure (capital F)

File: arability/app/models/gamer.rb [hunk 11]
```diff
@@ -62,6 +62,38 @@ def self.find_for_database_authentication(gamer_conditions)
...
+  #   removes the user from the admins
+  # Author:
+  #   Karim el naggar
+  # @params:
+  #   user: the user to be converted
+  # returns:
+  #   success:
+  #     the user is now not an admin
+  #   failure:
+  #     none
```
Comment (AmrARaouf) [inline arability/app/models/gamer.rb]: check comment above

**Comment by AmrARaouf** [comment]
@karimngr tests verified

**Comment by AmrARaouf** [comment]
@karimngr check your docs

**Comment by ghost** [comment]
@AmrARaouf  please reivew docs

**Comment by AmrARaouf** [comment]
@karimngr docs verified

**Comment by omar-mohyeldin** [comment]
@karimngr code verified

**Comment by Mirna-Yacout** [comment]
please start block with start of arrow =D

**Comment by ghost** [comment]
@Mirna-Yacout please review UML

**Comment by ghost** [comment]
@Mirna-Yacout  @m0hamed  done review

**Comment by ghost** [comment]
@Mirna-Yacout  @m0hamed  please review

**Comment by Mirna-Yacout** [comment]
verified

**Pull request merged**